examples/squeezer_bin/squeezer_bin/mod.rs

```diff
@@ -0,0 +1,45 @@
+mod imp;
+
+use gtk::glib;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+
+glib::wrapper! {
+    pub struct SqueezerBin(ObjectSubclass<imp::SqueezerBin>) @extends gtk::Widget;
+}
+
+impl SqueezerBin {
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Self {
+        glib::Object::new(&[]).unwrap()
+    }
+
+    pub fn keep_aspect_ratio(&self) -> bool {
+        self.imp().keep_aspect_ratio.get()
+    }
+
+    pub fn set_keep_aspect_ratio(&self, keep_aspect_ratio: bool) {
+        self.imp().keep_aspect_ratio.set(keep_aspect_ratio);
+        self.queue_resize();
+    }
+
+    fn get_child_size(&self) -> ((i32, i32), (i32, i32)) {
+        let (horizontal_minimal, horizontal_natural, _, _) = self
+            .first_child()
+            .unwrap()
+            .measure(gtk::Orientation::Horizontal, -1);
+        let (vertical_minimal, vertical_natural, _, _) = self
+            .first_child()
+            .unwrap()
+            .measure(gtk::Orientation::Vertical, -1);
+
+        (
+            (horizontal_minimal, horizontal_natural),
+            (vertical_minimal, vertical_natural),
+        )
+    }
+
+    pub fn imp(&self) -> &imp::SqueezerBin {
```

Comment:
is this needed? there is already a self.imp() provided for you.

Reply:
Done